machine/qemu/v1alpha1.go

```diff
@@ -104,6 +104,44 @@ func (service *machineV1alpha1Service) Create(ctx context.Context, machine *mach
 		return machine, fmt.Errorf("unsupported QEMU version: %s: please upgrade to a newer version", qemuVersion.String())
 	}
 
+	// Determine the QEMU machine type to use
+	qemuAccels, err := GetQemuMachineAccelFromBin(ctx, bin)
+	if err != nil {
+		return machine, err
+	}
+
+	if machine.Spec.Emulation {
+		emulation := false
+		for _, accel := range qemuAccels {
+			if accel == QemuMachineAccelTCG {
+				emulation = true
+				break
+			}
+		}
+
+		if !emulation {
+			return machine, fmt.Errorf("emulation requested but TCG not available")
+		}
+	} else {
+		var requestedAccel QemuMachineAccelerator
+		platform := false
+		if machine.Spec.Platform == "qemu" {
+			requestedAccel = QemuMachineAccelKVM
+		} else {
+			requestedAccel = QemuMachineAccelerator(machine.Spec.Platform)
```

Comment:
I understand the goal here but i don't think this will work because it will be filtered out at a much higher level before it reaches the qemu implementation.